addons/payment/tests/test_payment_acquirer.py

```diff
@@ -0,0 +1,66 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import tagged
+
+from odoo.addons.payment.tests.common import PaymentCommon
+
+
+@tagged('-at_install', 'post_install')
+class TestPaymentAcquirer(PaymentCommon):
+
+    def test_unpublished_acquirer_not_compatible_for_public_user(self):
+        self.acquirer.published = False
+        self.env = self.env(user=self.env.ref('base.public_user'))
+
+        acquirers_public_user = self.env['payment.acquirer'].sudo()._get_compatible_acquirers(
+            company_id=self.company.id,
+            partner_id=self.partner.id,
+        )
+        self.assertNotIn(self.acquirer, acquirers_public_user)
+
+    def test_unpublished_acquirer_not_compatible_for_portal_user(self):
+        self.acquirer.published = False
+        self.env = self.env(user=self.portal_user)
+
+        acquirers_portal_user = self.env['payment.acquirer'].sudo()._get_compatible_acquirers(
+            company_id=self.company.id,
+            partner_id=self.partner.id,
+        )
+        self.assertNotIn(self.acquirer, acquirers_portal_user)
+
+    def test_unpublished_acquirer_compatible_for_internal_user(self):
```

Comment:
This test is simpler than the first (merged) one so it should be moved before it.